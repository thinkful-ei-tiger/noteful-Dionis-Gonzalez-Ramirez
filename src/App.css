html {
  background-color: rgba(128, 128, 128, 0.137);
  min-width: 300px;
}

html, body, #root {
  height: 100%;
}

body {
  overflow: hidden;
  height: 100%;
  margin: 0;
}

main {
  display: flex;
  flex-wrap: wrap;
  box-sizing: border-box;
  height: 100%;
  margin: 0;
}

a {
  text-decoration: none;
}

a, button {
  cursor: pointer;
}

@media all and (max-width: 600px) {
  main {
    flex-direction: column;
    flex-wrap: nowrap;
  }
}
