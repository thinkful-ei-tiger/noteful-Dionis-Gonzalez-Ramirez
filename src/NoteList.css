.note-list {
  box-sizing: border-box;
  height: 100%;
  width: 70%;
  padding: 15px 30px;
  overflow-y: auto;
}

#add-note {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 35%;
  margin: 20px;
  height: 2.5em;
  border-radius: 5px;
  background-color: rgba(0, 170, 248, 0.781);
  border: solid;
  border-width: 1.5px;
  border-color: rgb(139, 131, 248);
  color: black;
  font-size: 18px;
}

#add-note:hover {
  background-color: rgb(231, 197, 2);
}

@media all and (max-width: 600px) {
  .note-list {
    display: flex;
    flex-direction: column;
    width: 100%;
    padding: 0 25px;
  }
}