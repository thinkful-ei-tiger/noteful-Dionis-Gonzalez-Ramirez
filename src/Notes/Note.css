@keyframes editingNoteTitle {
  0% {color: rgba(0, 170, 248, 0.507);}
  50% {color: rgba(255, 255, 255, 0.295); text-shadow: 2px 2px rgba(255, 0, 0, 0.521);}
  100% {color: rgba(0, 170, 248, 0.507);}
}

.note-preview {
  display: flex;
  flex-direction: column;
  box-sizing: border-box;
  min-height: 200px;
  border: solid;
  border-width: 1px;
  border-color: black;
  border-radius: 5px;
  padding: 10px 75px 10px 0;
  border-bottom-width: 0px;
  background-color: rgba(0, 170, 248, 0.507);
  text-decoration: none;
  font-size: 22px;
  font-weight: bold;
  text-shadow: 1px 1px rgb(255, 255, 255);
  color: rgba(0, 0, 0, 0.938);
  z-index: 0;
  cursor: pointer;
  position: relative;
}

.note-preview a[contenteditable='true'] {
  animation-name: editingNoteTitle;
  animation-duration: 0.5s;
  animation-iteration-count: infinite;
}

.note-preview a {
  display: flex;
  flex-direction: column;
  display: block;
  box-sizing: border-box;
  color: rgb(59, 49, 100);
  width: calc(100% - 35px);
  height: 3em;
  padding: 25px 0 20px 35px;
}

.note-preview p {
  text-shadow: none;
  font-size: 14px;
  font-weight: normal;
  color: rgb(80, 80, 80);
  cursor: pointer;
}

.note-preview:hover {
  background-color: rgba(0, 170, 248, 0.781)
}

.note-preview button {
  align-self: flex-end;
  justify-self: flex-start;
  border: solid;
  border-width: 2px;
  border-radius: 5px;
  font-size: 16px;
  width: 100px;
  padding: 5px 0;
  box-shadow: 2px 2px gray;
  margin: 10px 0;
  z-index: 2;
}

i {
  color: rgb(99, 97, 97);
}

.cancel-button {
  z-index: 3;
}

.buttons {
  display: flex;
  justify-content: space-around;
  margin-right: -25px;
}

@media all and (max-width: 600px) {
  .note-preview {
    text-align: center;
    padding: 0;
  }

  .note-preview button {
    margin-left: 15px;
    align-self: center;
  }

  .buttons {
    display: flex;
    flex-direction: column;
    align-items: center;
    box-sizing: border-box;
    width: 100%;
    /* margin-bottom: 15px; */
    padding-right: 20px;
  }
}
