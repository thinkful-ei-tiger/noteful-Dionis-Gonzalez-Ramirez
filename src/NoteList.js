import React from 'react';
import data from './data';
import Note from './Note'
import {Link} from 'react-router-dom';
import ErrorPage from './ErrorPage'
import './NoteList.css'

class NoteList extends React.Component {
  render() {
    const folderID = this.props.match.params.folder || '';
    const notes = data.notes.filter(note => note.folderId === folderID) || [];
    const noteExists = notes[0] !== undefined;
    const folderExists = data.folders.find(folder => folder.id === folderID) !== undefined

    return (
      (!noteExists && !folderExists)
      ? <ErrorPage />
      :
      <div className='note-list'>
        {
          notes.map(note => {
            return (
              <Note
                folderID={folderID}
                note={note}
                key={this.props.match.params.note}
              />
            )
          })
        }
        <Link id='add-note'to='/'>Add Note</Link>
      </div>
    )
  }
}

export default NoteList;