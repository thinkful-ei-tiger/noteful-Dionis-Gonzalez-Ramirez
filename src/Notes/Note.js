import React from 'react';
import {Link, withRouter} from 'react-router-dom';
import PropTypes from 'prop-types'
import NotesContext from '.././NotesContext'
import './Note.css'

class Note extends React.Component {
  state = {
    previousValues: {...this.props.note}
  }

  static contextType = NotesContext;

  static propTypes = {
    notes: PropTypes.array.isRequired,
    note: PropTypes.object.isRequired,
    key: PropTypes.string,
    folderID: PropTypes.string
  }

  editTitle = (newName) => {
    const noteID = 
      (this.props.note.id === undefined)
      ? this.props.match.params.note
      : this.props.note.id
    let found = this.props.notes.find(note => note.id === noteID)
    found = (found === undefined) ? {} : found;
    Object.assign(found, {name: newName})
  }
  cancel = () => {
  let found = this.props.notes.find(note => note.id === this.props.note.id)
  found = (found === null || found === undefined) ? {} : found;
  Object.assign(found, this.state.previousValues)
  this.props.history.push(`/folders/${this.props.note.folderId}`);
  }
  render() {
    const rawDate = 
      (this.props.match.params.note)
      ? (new Date()).toString()
      : new Date(this.props.note.modified).toString();
    const date =  rawDate.split('GMT')[0];
    const folderID = 
      [this.props.folderID, this.props.note.folderId, this.props.match.params.folder]
      .find(folder => folder !== undefined)
    const noteID =
      [this.props.note.id, this.props.match.params.note]
      .find(note => note !== undefined)
    return (
        <div className='note-preview'>
          <Link
            to={`/folders/${folderID}/notes/${noteID}`}
            key={this.props.note.id}
            suppressContentEditableWarning="true"
            onKeyDown={(evt) => (evt.which === 13) ? evt.target.blur() : null}
            onBlur={(evt) => {
              evt.target.contentEditable = false;
              this.editTitle(evt.target.innerText.split('\n')[0])
            }}
            onClick={(evt) => (evt.target.contentEditable === true) ? evt.target.toggle('note-title-selected') : evt.target.contentEditable = true}
          >
            {this.props.note.name}
          </Link>
          <div className='buttons'>
          <p contentEditable={false}><i>Modified on {date}</i></p>
          <button onClick={() => this.context.deleteNote(noteID)}>Delete</button>
          <button
            className='cancel-button'
            onClick={this.cancel}
            style={{
              display:
                (`/folders/${folderID}/notes/${noteID}` === this.props.match.url)
                ? 'block'
                : 'none'
            }}
          >Cancel</button>
          </div>
        </div>
    )
  }
}

export default withRouter(Note);