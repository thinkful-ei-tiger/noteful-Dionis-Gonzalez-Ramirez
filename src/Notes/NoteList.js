import React from 'react';
import {withRouter} from 'react-router-dom'
import PropTypes from 'prop-types'
import Note from './Note'
import ErrorPage from '../DefaultPages/ErrorPage'
import NotesContext from '.././NotesContext'
import './NoteList.css'

class NoteList extends React.Component {
  state = this.props.state;

  static contextType = NotesContext;

  static propTypes = {
    addNote: PropTypes.func.isRequired,
    deleteNote: PropTypes.func.isRequired,
    state: PropTypes.object.isRequired
  }

  render() {
    const folderID = this.props.match.params.folder;
    const notes = this.props.state.notes.filter(note => note.folderId === folderID);
    const note = notes[0]
    return (
      (this.props.state.folders.find(folder => folder.id === folderID) === undefined)
      ? <ErrorPage />
      :
      <div className='note-list'>
        {
          notes.map(note => {
            return (
              <Note
                folderID={folderID}
                note={note}
                key={note.id}
                notes={this.props.state.notes}
              />
            )
          })
        }
        <button
          onClick={() => this.context.addNote(folderID)}
          id='add-note'
        >Add Note</button>
      </div>
    )
  }
}

export default withRouter(NoteList);