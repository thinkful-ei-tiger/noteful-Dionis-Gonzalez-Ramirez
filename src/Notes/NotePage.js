import React from 'react';
import {withRouter} from 'react-router-dom'
import PropTypes from 'prop-types'
import Note from './Note';
import NotesContext from '.././NotesContext'
import './NotePage.css'

class NotePage extends React.Component {
  static contextType = NotesContext;

  static propTypes = {
    notes: PropTypes.array.isRequired,
    deleteNote: PropTypes.func.isRequired
  }

  editContent = (newContent) => {
    let found = this.props.notes.find(note => note.id === this.props.match.params.note)
    found = (found === undefined) ? {} : found;
    found.content = newContent;
    found.modified = (new Date()).toISOString();
  }

  render = () => {
    let note = this.props.notes.find(note => note.id === this.props.match.params.note)
    note = (note === undefined) ? {} : note;
    return (
      <div className='note-page'>
        <Note
          note={note}
          notes={this.props.notes}
        />
        <div
          contentEditable
          onBlur={(evt) => this.editContent(evt.target.textContent)}
          suppressContentEditableWarning="true"
          className= {'note-content'}
        >
          {note.content}
        </div>
      </div>
    )
  }
}

export default withRouter(NotePage);