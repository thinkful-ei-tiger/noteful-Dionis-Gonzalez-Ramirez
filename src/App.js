import React from 'react';
import {Route} from 'react-router-dom';
import HomePage from './HomePage';
import NoteList from './NoteList';
import NotePage from './NotePage';
import './App.css'

function App() {
  return (
    <main className='App'>
      
      <Route path='/' component={HomePage} />
      <Route exact path='/:folder' component={NoteList} />
      <Route path='/:folder/:note' component={NotePage} />

    </main>
  );
}

export default App;
