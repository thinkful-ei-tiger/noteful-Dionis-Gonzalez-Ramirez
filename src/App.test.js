import React from 'react'
import ReactDOM from 'react-dom'
import {BrowserRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'
import {NoteList} from './Notes/NoteList'
import App from './App';

test('renders without crashing (smoke test)', () => {
  const div = document.createElement('div');
  ReactDOM.render(<BrowserRouter><App /></BrowserRouter>, div)
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the User Interface as expected (snapshot test)', () => {
  const tree = renderer
    .create(NoteList)
    .toJSON();
  expect(tree).toMatchSnapshot();  
  });

