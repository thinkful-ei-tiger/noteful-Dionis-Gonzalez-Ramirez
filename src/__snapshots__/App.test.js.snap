// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the User Interface as expected (snapshot test) 1`] = `null`;
